src/localplayer.cpp

```diff
@@ -179,25 +180,22 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 		If sneaking, keep in range from the last walked node and don't
 		fall off from it
 	*/
-	if(control.sneak && m_sneak_node_exists &&
+	if (control.sneak && m_sneak_node_exists &&
 			!(fly_allowed && g_settings->getBool("free_move")) && !in_liquid &&
-			physics_override_sneak)
-	{
-		f32 maxd = 0.5*BS + sneak_max;
+			physics_override_sneak) {
+		f32 maxd = 0.5 * BS + sneak_max;
 		v3f lwn_f = intToFloat(m_sneak_node, BS);
 		position.X = rangelim(position.X, lwn_f.X-maxd, lwn_f.X+maxd);
 		position.Z = rangelim(position.Z, lwn_f.Z-maxd, lwn_f.Z+maxd);
 
-		if(!is_climbing)
-		{
-			f32 min_y = lwn_f.Y + 0.5*BS;
-			if(position.Y < min_y)
-			{
-				position.Y = min_y;
+		if (!is_climbing) {
+			// Move up if necessary
+			f32 new_y = (lwn_f.Y - 0.5 * BS) + m_sneak_node_bb_ymax;
+			if (position.Y < new_y)
+				position.Y = new_y;
 
-				if(m_speed.Y < 0)
-					m_speed.Y = 0;
-			}
+			if (m_speed.Y < 0)
```

Comment:
This block now gets executed even when position.Y >= min_y.  Was this intentional?